content/4_Gold/PURS.mdx

```diff
@@ -579,6 +582,74 @@ int main() {
 
 </CPPSection>
 
+<JavaSection>
+
+```java
+import java.util.*;
+import java.io.*;
+
+public class DynamicRangeSumQueries {
+	public static void main(String[] args) {
+		Kattio io = new Kattio();
+		int n = io.nextInt();
+		int q = io.nextInt();
+
+		// Note: we are using 1-based indexing for the BIT to function properly
+		// "arr" stores the value of each item in the array for convenient use
+		BinaryIndexTree bit = new BinaryIndexTree(n);
```

Comment:
why make it a class?

Reply:
There isn't a real reason, I mainly did it cause I'm used to OOP but I can change it

Reply:
what's wrong with using a class?

Reply:
I'd say change it real quick to match the c++ version